sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/AuthClient.java

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.security.keyvault.jca;
+
+import com.azure.security.keyvault.jca.rest.OAuthToken;
+
+import java.util.HashMap;
+import java.util.logging.Logger;
+
+import static java.util.logging.Level.FINER;
+
+/**
+ * The REST client specific to getting an access token for Azure REST APIs.
+ */
+class AuthClient extends DelegateRestClient {
+
+    /**
+     * Stores the Client ID fragment.
+     */
+    private static final String CLIENT_ID_FRAGMENT = "&client_id=";
+
+    /**
+     * Stores the Client Secret fragment.
+     */
+    private static final String CLIENT_SECRET_FRAGMENT = "&client_secret=";
+
+    /**
+     * Stores the Grant Type fragment.
+     */
+    private static final String GRANT_TYPE_FRAGMENT = "grant_type=client_credentials";
+
+    /**
+     * Stores the Resource fragment.
+     */
+    private static final String RESOURCE_FRAGMENT = "&resource=";
+
+    /**
+     * Stores the OAuth2 token base URL.
+     */
+    private static final String OAUTH2_TOKEN_BASE_URL = "https://login.microsoftonline.com/";
```

Comment:
Is this intended to work against clouds other than Azure Public Cloud? If so this would need to be configurable.